optuna/integration/cma.py

```diff
@@ -312,8 +319,12 @@ def __init__(
                 lows.append(0 - 0.5 * dist.q)
                 highs.append(r + 0.5 * dist.q)
             elif isinstance(dist, IntUniformDistribution):
-                lows.append(dist.low - 0.5)
-                highs.append(dist.high + 0.5)
+                lows.append(dist.low - 0.5 * dist.step)
+                highs.append(dist.high + 0.5 * dist.step)
+            elif isinstance(dist, IntLogUniformDistribution):
+                # TODO(toshihikoyanase): Shiting 0.5 is not sufficient if step > 1.
+                lows.append(self._to_cma_params(search_space, param_name, dist.low - 0.5))
+                highs.append(self._to_cma_params(search_space, param_name, dist.high + 0.5))
```

Comment:
Your second concern is reasonable: instead of expanding 0.5, we should expand 0.5*step. As you say, I think that `IntLogUniformDistrbituion` uses 0.5 to ensure `low` positive, but if we just want to ensure it, how about doing the following? I think the `optuna.samplers.CmaEsSampler` can be fixed as well.

```suggestion
                # TODO(toshihikoyanase): Shiting 0.5 is not sufficient if step > 1.
                lows.append(self._to_cma_params(search_space, param_name, max(0, dist.low - 0.5 * dist.step)))
                highs.append(self._to_cma_params(search_space, param_name, dist.high + 0.5 * dist.step))
```

Reply:
This change still has the problem of bias in discretization when `low - 0.5 * step` is negative. A possible solution is to move `[low-0.5*step, high+0.5*step]` parallel to `[1, high - low + step + 1]` for sampling when `low - 0.5 * step` is negative, and then undo it later.  However, this is a change that has to be made to other samplers at the same time, and I think it's outside the scope of this PR.

Reply:
I'm sorry for the delayed response.
Your new PR (#1329) will prohibit the simultaneous use of `step` and `log`, and we can keep the `master`'s implementation. So, I'll revert the change.